version: '3.1'
services:
  db:
      image: postgres
      environment:
        - POSTGRES_DB=postgres
        - POSTGRES_USER=postgres
        - POSTGRES_PASSWORD=Brian568
      healthcheck:
        test: [ "CMD-SHELL", "pg_isready -U myuser" ]
        interval: 2s
        timeout: 2s
        retries: 5
      volumes:
            - data-volume:/var/lib/postgresql/data
  pgadmin:
    image: dpage/pgadmin4
    restart: always
    environment:
      PGADMIN_DEFAULT_EMAIL: 568@duke.edu
      PGADMIN_DEFAULT_PASSWORD: Brian568
    ports:
      - '5050:80'
    depends_on:
      db:
        condition: service_healthy
  ups-server:
    build: .
    command: python3 ups-service.py
    volumes:
      - '.:/code'
    ports:
      - '54321:54321'
    depends_on:
      db:
        condition: service_healthy
  world-server:
    build: .
    command: python3 world-service.py
    volumes:
      - '.:/code'
    depends_on:
      db:
        condition: service_healthy
  webapp:
    build: ./ups-webapp
    command: npm start
    ports:
      - '4000:4000'
    depends_on:
      db:
        condition: service_healthy
volumes:
  data-volume: null